src/bread-n-butter.ts

```diff
@@ -307,6 +307,26 @@ export function match(regexp: RegExp): Parser<string> {
   });
 }
 
+/** A tuple of parsers */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type ManyParsers<A extends any[]> = {
+  [P in keyof A]: Parser<A[P]>;
+};
+
+/** Parse all items, returning their values in the same order. */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function all<A extends any[]>(...parsers: ManyParsers<A>): Parser<A> {
+  // TODO: This could be optimized with a custom parser, but I should probably add
+  // benchmarking first to see if it really matters enough to rewrite it
+  return parsers.reduce((acc, p) => {
+    return acc.chain((array) => {
+      return p.map((value) => {
+        return [...array, value];
+      });
+    });
+  }, ok<A[]>([]));
```

Comment:
This is a _logical_ error, because `A` is already an array. The real parameter of `ok` should be `A`, but it does not work because initialized with an empty array, which is not an `A`. I think that the better is to not type `ok` at all:

```typescript
…
}, ok([]);
```

but if you really want to provide a type, you probably can do:

```typescript
…
}, ok<A>([] as A);
```

Reply:
Good point, I'll leave it without types :)